### PR TITLE
Added the basic Wavefunction class

### PR DESCRIPTION
**Pull request by psavery**
This should be complete unless new issues arise.

The Wavefunction class in #4 is not the most updated one.

### CODE REVIEW

**Comment by dgasmith** [comment]
This is in two places, you are going to have to merge this history..

File: quest/wavefunction.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+"""
+
+import psi4
+
+
+class WaveFunction(object):
+    """
+    Basic wave function class
+
+    Members:
```
Comment (dgasmith) [inline quest/wavefunction.py]: `Wavefunction` is a single word so do not add a capitol in the middle. Please see the NumPy style docs [here](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt).

File: quest/wavefunction.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Initialize the WaveFunction class.
+
+        Args:
+            mol: a Molecule class from the QuESt repository. Needs to
+                 have a basis set at least.
+        """
+
+        # Set whichever options you would like. The only default is the
+        # basis set name.
+        self.options = {'basis_name': mol.bas.name}
```
Comment (dgasmith) [inline quest/wavefunction.py]: This will be `mol.basisset().name()`.

**Review by dgasmith — CHANGES_REQUESTED** [review]

**Comment by psavery** [comment]
Yes, we did not intend for the DIiS group to submit our wave function class as part of their PR. They did that by accident.

I will help them remove it from their PR

File: quest/wavefunction.py (hunk 2 above)
Comment (psavery) [reply quest/wavefunction.py]: Okay. I can add it, but they will need to add a "basisset()" function in their molecule class for that to work:

https://github.com/MolSSI-SSS/SSS_2017_QuESt/blob/master/quest/molecule.py

Comment (dgasmith) [reply quest/wavefunction.py]: Ah, its from the `molecule`, never mind then!

Comment (kml5480) [reply quest/wavefunction.py]: I think this should be calling psi4.core.BasisSet.name(bas) but I might be mistaken!

**Review by dgasmith — APPROVED** [review]

**Pull request merged**